File: DSA/Deeptansh-nagarr/day 8/question2.java
import java.util.LinkedList;
import java.util.Queue;
import java.util.Scanner;

class StackUsingTwoQueues {
    private Queue<Integer> q1 = new LinkedList<>();
    private Queue<Integer> q2 = new LinkedList<>();

    public void push(int x) {
        q2.add(x);
        while (!q1.isEmpty()) {
            q2.add(q1.poll());
        }
        Queue<Integer> temp = q1;
        q1 = q2;
        q2 = temp;
    }

    public int pop() {
        if (q1.isEmpty()) {
            System.out.println("Stack is empty");
            return -1;
        }
        return q1.poll();
    }

    public int top() {
        if (q1.isEmpty()) {
            System.out.println("Stack is empty");
            return -1;
        }
        return q1.peek();
    }
}

public class Main {
    public static void main(String[] args) {
        StackUsingTwoQueues stack = new StackUsingTwoQueues();
        Scanner scanner = new Scanner(System.in);

        while (true) {
            System.out.println("1. Push");
            System.out.println("2. Pop");
            System.out.println("3. Top");
            System.out.println("4. Exit");
            System.out.print("Enter your choice: ");
            int choice = scanner.nextInt();

            switch (choice) {
                case 1:
                    System.out.print("Enter element to push: ");
                    int element = scanner.nextInt();
                    stack.push(element);
                    break;
                case 2:
                    int poppedElement = stack.pop();
                    if (poppedElement != -1) {
                        System.out.println("Popped element: " + poppedElement);
                    }
                    break;
                case 3:
                    int topElement = stack.top();
                    if (topElement != -1) {
                        System.out.println("Top element: " + topElement);
                    }
                    break;
                case 4:
                    System.exit(0);
                    break;
                default:
                    System.out.println("Invalid choice");
            }
        }
    }
}

File: DSA/Advik/Day9/question2.cpp
#include<bits/stdc++.h>

using namespace std;

bool compare(char a){
    vector<char> ar={'(','{','['};
    for(int i=0;i<=ar.size()-1;i++){
        if(a==ar[i]){
            return true;
        } 
    }
    return false;
}

    char opp(char top){
    if(top=='}'){
        return '{';
    }
    if(top==')'){
        return '(';
    }
    if(top==']'){
        return '[';
    }
    return 'l';
}
    
bool isValid(string s) {
        stack<char> k;
        for(int i=0;i<=s.size()-1;i++){
            if(compare(s[i])){
                k.push(s[i]);
            }
            else if(k.empty()){return false;}
            else if(opp(s[i])==k.top()){
                
                k.pop();
            }
            else{return false;}
            
        }
        if(k.empty()){
            return true;
        }
        else{return false;}


    }





int main(){
    string str="{[()]}";
    if(isValid(str)) cout<<"YES";
    else cout<<"NO";

}